### PR TITLE
Copy project icons to Gradle project directory during Android Custom Build.

### PR DESCRIPTION
**Pull request by amanj120**
Forward port of #39997 

@m4gr3d tagging you for visibility

### CODE REVIEW

**Comment by akien-mga** [comment]
For the reference, a proposal that describes what you're aiming to implement and the various changes that are necessary for this would have been welcome. As it stands, there are a number of PRs for both `3.2` and `master` which all implement some steps without a clear overview.

I trust @m4gr3d to get the necessary overview and thus defer to his assessment for merging changes into `master`. For the `3.2` branch, I'll wait for everything to be implemented and tested. I'd suggest focusing on `master` and eventually make a PR with cherrypicks of all relevant commits (+ potential subsequent fixes) for the `3.2` branch when the feature has been fully implemented.

**Comment by amanj120** [comment]
@akien-mga Thank you for the advice. I will focus my attention on creating a fully tested and implemented Android App Bundle exporting option on the master branch. Once it works, I'll create a single, comprehensive, PR that cherry-picks all the changes to the 3.2 branch. 

However, a single PR to backport all the changes from master to 3.2 will be very large (on the order of 500+ lines), will that be acceptable?

Also, should I close all the relevant PRs on the 3.2 branch that are open right now (#39864, #39998, #40182, #39997)?

**Comment by akien-mga** [comment]
> However, a single PR to backport all the changes from master to 3.2 will be very large (on the order of 500+ lines), will that be acceptable?

As long as the commits are kept separate as they are in the current PRs (and easily identified as backports of the corresponding PRs in the `master` branch), it should be good.

But let's see what @m4gr3d thinks about it, if he prefers to have the individual `3.2` PRs kept and merged as is, that's OK too.

We discussed last week with @m4gr3d that to allow a better integration of new features in `3.2` without compromising the stability of the already released branch, we'd probably merge new features in specific feature branches (e.g. `3.2-android-app-bundle`) with their own sets of official binaries to get some testing from the community before integrating it in the main `3.2` branch for the next maintenance release.

So actually we could do that now, we can create a dedicated branch and merge your existing `3.2` PRs in that branch, which can later be merged in `3.2`. If that sounds good to you, I'll create that branch and we can edit the existing PRs to target that branch as base branch.

**Comment by amanj120** [comment]
> So actually we could do that now, we can create a dedicated branch and merge your existing `3.2` PRs in that branch, which can later be merged in `3.2`. If that sounds good to you, I'll create that branch and we can edit the existing PRs to target that branch as base branch.

I think that's a good solution.

**Comment by akien-mga** [comment]
Alright, here's the branch: https://github.com/godotengine/godot/tree/3.2-android-app-bundle

I merged #39864 and #39998.

#40182 depends on #40260 being merged first (and updated to match any changes).
And #39997 depends on this PR being merged first.

File: platform/android/export/export.cpp [hunk 1]
```diff
@@ -29,6 +29,7 @@
 /*************************************************************************/
 
 #include "export.h"
+#include "gradle_export_util.h"
```
Comment (amanj120) [inline platform/android/export/export.cpp:32]: this is something that is fixed by #40422, which needs to be merged before this

File: platform/android/export/export.cpp [hunk 2]
```diff
@@ -2003,7 +2065,9 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 
 		EditorProgress ep("export", "Exporting for Android", 105, true);
 
-		if (bool(p_preset->get("custom_template/use_custom_build"))) { //custom build
+		bool use_custom_build = bool(p_preset->get("custom_template/use_custom_build"));
```
Comment (amanj120) [inline platform/android/export/export.cpp:2068]: This is also another change made in #40422

File: platform/android/export/gradle_export_util.h [hunk 3]
```diff
@@ -94,7 +94,7 @@ Error store_string_at_path(const String &p_path, const String &p_data) {
 // This method will be called ONLY when custom build is enabled.
 Error rename_and_store_file_in_gradle_project(void *p_userdata, const String &p_path, const Vector<uint8_t> &p_data, int p_file, int p_total) {
 	String dst_path = p_path.replace_first("res://", "res://android/build/assets/");
-	Error err = store_file_at_path(dst_path, p_data, Z_NO_COMPRESSION);
+	Error err = store_file_at_path(dst_path, p_data);
```
Comment (amanj120) [inline platform/android/export/gradle_export_util.h:97]: Also in #40422

**Review by m4gr3d — APPROVED** [review]
Looks great!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**